### ISSUE TITLE
metrics are not evaluated if loss is None

### ISSUE THREAD

**Issue by tik0** [keras-team/keras#12016]
- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Using the [VAE example from Keras](https://github.com/keras-team/keras/blob/master/examples/variational_autoencoder.py), one can easily reproduce the effect that any metric defined in the `model.compile` call will not be evaluated if the `loss` argument is None. For instance, substituting `vae.compile(optimizer='adam')` in [line 188](https://github.com/keras-team/keras/blob/master/examples/variational_autoencoder.py#L188) with 
```
    def kl_loss_fun(y_true, y_pred):
        return kl_loss
    def reconstruction_loss_fun(y_true, y_pred):
        return reconstruction_loss
    vae.compile(optimizer='adam', metrics=[kl_loss_fun, reconstruction_loss])
```
shows no output of the metrics when `fit` is called.

I found out that, if [compile](https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L37) is called, that the metrics are just skipped if the corresponding loss per output is None in [line 443](https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L443). Removing the conditional fixes the issue!

**So the main question is, why is the calculation of the metric conditioned on the losses at all? If the user demands it explicitly it should not depend on the loss which is given in the compile call.**

**Comment by gabrieldemarmiesse** [comment]
This seems very strange. Fix welcome. Can you provide a small and standalone script which would serve as a test to reproduce the issue? Thanks!

**Comment by tik0** [comment]
@sakvaua this sounds a bit overengineered. It would be more straight forward to evaluate the metric in a callback instead of defining a model with a mockup-loss to actually evaluate the desired metric. Anyway, the question remains concerning [line 443](https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L443): Wouldn't removing the conditional fix this issue?

**Comment by tik0** [comment]
Just found an open issue concerning this topic: #9459.
I will therefore close this issue!
